com.unity.shadergraph/Editor/ShaderFoundry/Utilities/Extensions.cs

```diff
@@ -191,11 +191,11 @@ internal static void MergeTypesAndFunctions(this Block.Builder builder, Block bl
             foreach (var item in block.ReferencedTypes)
                 builder.AddReferencedType(item);
             foreach (var item in block.Types)
-                builder.AddReferencedType(item);
+                builder.AddType(item);          // NOTE: parent block will be wrong...
```

Comment:
This is wrong. The new block doesn't own the type of the old block, it's a reference type. Why did you need to change this?

Reply:
Yeah, AddType is "owned" types, so they're in the block's namespace. Referenced types are any that you may use, so we can track them for knowing what types to emit.